src/app/modules/main/module.nim

```diff
@@ -1411,6 +1413,9 @@ method onStatusUrlRequested*[T](self: Module[T], action: StatusUrlAction, commun
     of StatusUrlAction.OpenCommunity:
       let item = self.view.model().getItemById(communityId)
       if item.isEmpty():
+        if self.controller.getCommunityById(communityId).id != "":
+          self.controller.spectateCommunity(communityId)
+          return
```

Comment:
This saves us a useless fetch. If we already know the community, just spectate it.